<?xml version="1.0" ?>
<launch>

	<arg name="paused" default="false"/>
	<arg name="use_sim_time" default="true"/>
	<arg name="gui" default="true"/>
	<arg name="headless" default="false"/>
	<arg name="debug" default="false"/>
	<arg name= "x" default= "0.0"/>
	<arg name= "y" default= "0.0"/>
	<arg name= "z" default= "0.25"/>
	<arg name= "R" default= "0.0"/>
	<arg name= "P" default= "0.0"/>
	<arg name= "Y" default= "0.0"/>
	
	<include file="$(find technical_project)/launch/robot_description.launch"/>
	<env name="GAZEBO_MODEL_PATH" value="$(find technical_project)/models:$(optenv GAZEBO_MODEL_PATH)" />

	<include file="$(find gazebo_ros)/launch/empty_world.launch">
		<arg name="world_name" value="$(find technical_project)/worlds/scenario.world"/>
		<arg name="debug" value="$(arg debug)" />
		<arg name="gui" value="$(arg gui)" />
		<arg name="paused" value="$(arg paused)"/>
		<arg name="use_sim_time" value="$(arg use_sim_time)"/>
		<arg name="headless" value="$(arg headless)"/>
	</include>

	<param name="robot_description" command="$(find xacro)/xacro '$(find technical_project)/urdf/powerbot/powerbot.xacro'" /> 
	<rosparam file ="$(find technical_project)/config/powerbot_controller.yaml" command="load"/> 

	<node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
	args="-urdf -model diff_wheeled_robot -param robot_description -x $(arg x) -y $(arg y) -z $(arg z) -R $(arg R) -P $(arg P) -Y $(arg Y)"/>
	<node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="joint_state_controller 
	left_wheel_velocity_controller right_wheel_velocity_controller"/> 

	<arg name="markerSize"      default="2.00"/>   
	<arg name="ref_frame"       default=""/> 

	<node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher">
		<remap from="/camera_info" to="/camera/camera_info" />
		<remap from="/image" to="/camera/image_raw" />
		<param name="image_is_rectified" value="True"/>
		<param name="marker_size"        value="$(arg markerSize)"/>
		<param name="reference_frame"    value="$(arg ref_frame)"/>  
	</node>

</launch>
